mapie/estimator/interface.py

```diff
@@ -0,0 +1,340 @@
+        Otherwise an error would be raised.
+
+        The Jackknife+ interval can be interpreted as an interval around the
+        median prediction, and is guaranteed to lie inside the interval,
+        unlike the single estimator predictions.
+
+        When the cross-validation strategy is ``Subsample`` (i.e. for the
+        Jackknife+-after-Bootstrap method), this function is also used to
+        aggregate the training set in-sample predictions.
+
+        If ``cv`` is ``"prefit"`` or ``"split"``, ``agg_function`` is ignored.
+
+        By default ``"mean"``.
+
+    verbose: int
+        The verbosity level, used with joblib for multiprocessing.
+        The frequency of the messages increases with the verbosity level.
+        If it more than ``10``, all iterations are reported.
+        Above ``50``, the output is sent to stdout.
+
+        By default ``0``.
+
+    random_state: Optional[Union[int, RandomState]]
+        Pseudo random number generator state used for random sampling.
+        Pass an int for reproducible output across multiple function calls.
+
+        By default ``None``.
+
+    Attributes
+    ----------
+    single_estimator_: sklearn.RegressorMixin
+        Estimator fitted on the whole training set.
+
+    estimators_: list
+        List of out-of-folds estimators.
+
+    k_: ArrayLike
+        - Array of nans, of shape (len(y), 1) if ``cv`` is ``"prefit"``
+          (defined but not used)
+        - Dummy array of folds containing each training sample, otherwise.
+          Of shape (n_samples_train, cv.get_n_splits(X_train, y_train)).
+    """
+    no_agg_cv_ = ["prefit", "split"]
+    no_agg_methods_ = ["naive", "base"]
+    fit_attributes = [
+        "single_estimator_",
+        "estimators_",
+        "k_",
+    ]
+
+    @staticmethod
+    def _fit_oof_estimator(
+        estimator: RegressorMixin,
+        X: ArrayLike,
+        y: ArrayLike,
+        train_index: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> RegressorMixin:
+        """
+        Fit a single out-of-fold model on a given training set.
+
+        Parameters
+        ----------
+        estimator: RegressorMixin
+            Estimator to train.
+
+        X: ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        y: ArrayLike of shape (n_samples,)
+            Input labels.
+
+        train_index: ArrayLike of shape (n_samples_train)
+            Training data indices.
+
+        sample_weight: Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If None, then samples are equally weighted.
+            By default ``None``.
+
+        Returns
+        -------
+        Tuple[RegressorMixin, NDArray, ArrayLike]
+
+        - [0]: RegressorMixin, fitted estimator
+        - [1]: NDArray of shape (n_samples_val,),
+          estimator predictions on the validation fold.
+        - [2]: ArrayLike of shape (n_samples_val,),
+          validation data indices.
```

Comment:
ERR: this docstring does not correspond to the current output of this method.

Reply:
Done